lib/plan_executor/executor.rb

```diff
@@ -0,0 +1,301 @@
+# frozen_string_literal: true
+
+# Used for $ERROR_INFO. This *must* be capitalized!
+require 'English'
+require 'json'
+require 'concurrent'
+require 'logging'
+require 'set'
+require 'bolt/result'
+require 'bolt/config'
+require 'bolt/notifier'
+require 'bolt/result_set'
+require 'bolt/puppetdb'
+
+module PlanExecutor
+  class Executor
+    attr_reader :noop, :transport
+    attr_accessor :run_as
+
+    def initialize(concurrency = 1,
+                   # XXX Are we supporting noop here?
+                   noop = nil,
+                   bundled_content: nil,
+                   load_config: true)
+      @bundled_content = bundled_content
+      @logger = Logging.logger[self]
+      @plan_logging = false
+      @load_config = load_config
+      @noop = noop
+      @run_as = nil
+      @pool = if concurrency > 0
```

Comment:
I don't think we need threading here because each puma thread has it's own executor(and instance of the app class) right?

Reply:
Assuming thats the case I think we can write a much much simpler executor implementation that just has an "orchestrator transport" and implements only the run_* functions which call the transport directly. All the threading and queueing is too support ssh/winrm etc.